### ISSUE TITLE
Allow unlimited characters in content.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [mastodon/mastodon#27433]
### Pitch

Do not limit the amount of characters possible in a post, a user's biography, or image description.

0.	**Post**

	Visibility issues can be solved by implementing a maximum initial post preview size, which, upon manual expansion, displays the entire post. The preview could additionally include a scrollbar so that to view additional content, expansion would be unnecessary.

0.	**Biography**

0.	**Image description**

	Image descriptions are viewed singularly, so no additional work to ensure visibility would be necessary.

### Motivation

0.	**Post**

	Per @volkriss at https://github.com/mastodon/mastodon/issues/12223#issuecomment-1328285237:
	
	<blockQuote>
	
	Keep in mind that ActivityPub supports longer compositions like Article and Page, so when talking about federation it wouldn't be Mastodon breaking things by providing such content. Rather, Mastodon can expect to deal with longer content incoming and needs to not break.
	
	The rest of this is personal opinion:
	
	I see doing away with twitter-style threads by lifting the cap as being one of the major contributions Mastodon can make toward improving social media. When people are sending out a dozen posts in a row because they can't fit what they have to say in a single post, that's a failure of the tool to do what they need to enable thoughtful expression. It's always amazing to me that people tolerate it.
	
	</blockQuote>

0.	**Biography**

	This biography form barely displays a biography. It's a tag-line at best. In order for a person's biography to be meaningful and prominent, it indeed must be reasonably concise, but a few paragraphs is verbose enough to explain well what a person's profile is about, whereas a few lines are not.

**Comment by tenebrios** [comment]
This is from Oct 2023
Nothing has changed 
Still 500 limit
Is it ever going to change?

**Comment by eishiya** [comment]
There will probably always be a limit for practical/technical reasons, but it doesn't have to be 500 characters. I would like to see the limit be configurable on a per-instance basis. This is currently doable with some manual tweaks to the database and it's how many Mastodon instances allow 1000 or more characters per post, but it would be much better if it was configurable via the admin interface and didn't get wiped with every Mastodon update. I suspect most Mastodon instances will want to keep a lowish limit, since Mastodon is really built around microblogging, and there are other federated services that may be more suitable for people who frequently make long posts.

I think bios should be kept brief, but since one is encouraged to put some tags in them and since one only gets 4 extra fields for links, pronouns, etc, I do think they could use some extra characters. Perhaps they should be displayed clipped if they're longer than the current limit, with the option to expand, similar to how long posts are currently displayed with a "read more".

The character limit for image descriptions already feels pretty good to me, and most of the very long descriptions I've seen have been much worse at being alt text than shorter ones. But, I feel that the problem of description quality is best solved by education, not by enforcing a short limit, so I'm not against an increase in the description character limit. For descriptions that transcribe long texts in the image, the current limit is particularly inadequate. While I don't want to encourage posts that are just screenshots from other sites, transcriptions of long texts are useful in other scenarios as well.

**Comment by RokeJulianLockhart** [comment]
> Perhaps they should be displayed clipped if they're longer than the current limit, with the option to expand, similar to how long posts are currently displayed with a "read more".

@eishiya, that's how Discourse renders long-form biographies.

**Comment by trwnh** [comment]
for bio/summary there is #21277 which comes at it from the other direction —- support a “short bio” and “long bio” separately

**Comment by mjankowski** [comment]
Maybe also related to descriptions - https://github.com/mastodon/mastodon/pull/32593

**Comment by RokeJulianLockhart** [comment]
> https://github.com/mastodon/mastodon/issues/27433#issuecomment-2502209968

Yeah, tangentially. However, that's a matter of accessibility, so they shan't be exposed unless the the images are maximised or disabled by the viewing user.

**Comment by fungilife** [comment]
This is so freaking ridiculous to CLOSE the issue that makes mastodon just as bad as X 

It is an insult to intelligence to limit ascii characters while uploading A/V files is unlimited till server crashes ...   

Why is mastodon trying so hard to look like a corporate anti-intelligent people medium who can't write or read more than 2-3 lines of giggles

**Comment by beedell-roke** [comment]
@fungilife, although I affirm your sentiment, this issue hasn't been closed. Which do you refer to?

**Comment by akulbe** [comment]
> This is so freaking ridiculous to CLOSE the issue that makes mastodon just as bad as X
> 
> It is an insult to intelligence to limit ascii characters while uploading A/V files is unlimited till server crashes ...
> 
> Why is mastodon trying so hard to look like a corporate anti-intelligent people medium who can't write or read more than 2-3 lines of giggles

I know you can modify `MAX_CHARS` in `live/app/validators/status_length_validator.rb` *if you control the instance* and restart your Mastodon processes, and it'll work.

<img width="354" height="336" alt="Image" src="https://github.com/user-attachments/assets/eaf2caa4-5a57-4f15-9a27-bcf59f7391c0" />

I'm not sure if that's the same thing you're asking for, or not.
